### PR TITLE
SQLCommands

### PR DESCRIPTION
**Pull request by Stanislava88**
In this PR i have solved the first task: creation methods to basic commands

### CODE REVIEW

File: src/com/clouway/sqlcommands/doc/schema.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE DATABASE university;
```
Comment (freekman) [inline src/com/clouway/sqlcommands/doc/schema.sql]: i think this is repeated with ```....jdbc/com/clouway/sqlcommands/doc/schema.sql```

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      preparedStatement.setInt(1, student.id);
+      preparedStatement.setString(2, student.first_name);
+      preparedStatement.setString(3, student.last_name);
+      preparedStatement.setInt(4, student.age);
+
+      preparedStatement.executeUpdate();
+    }
+  }
+
+  public Student findByID(int id) throws SQLException {
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: you have ```findById``` and ```selectAll``` eather use select or find not both

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    connection.close();
+  }
+
+  @Test
+  public void happyPath() throws Exception {
+    Student student = new Student(1, "Ivan", "Ivanov", 23);
+    repository.register(student);
+
+    List<Student> actual = repository.selectAll();
+    List<Student> expected = Lists.newArrayList(student);
```
Comment (freekman) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: use findById

File: src/com/clouway/sqlcommands/doc/schema.sql (hunk 1 above)
Comment (Stanislava88) [inline src/com/clouway/sqlcommands/doc/schema.sql]: fixed

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    assertThat(expected, is(actual));
+  }
+
+  @Test
+  public void registerMultipleStudent() throws Exception {//multiple
```
Comment (freekman) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: remove comments

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    repository.register(student2);
+
+    List<Student> actual = repository.selectAll();
+    List<Student> expected = Lists.newArrayList(student1, student2);
+
+    assertThat(expected, is(actual));
+  }
+
+  @Test
+  public void find() throws Exception {
```
Comment (freekman) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: findByID

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  @Test
+  public void update() throws Exception {
+    Student student = new Student(1, "Lilia", "Angelova", 24);
+
+    repository.register(student);
+    repository.update(new Student(1, "Lilia", "Angelova", 28));
+
+    List<Student> actual = repository.selectAll();
```
Comment (freekman) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: if you will use ```selectAll``` you should register one more student

Comment (Stanislava88) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: done

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java (hunk 5 above)
Comment (Stanislava88) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: done

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    try (PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM students WHERE id=?")) {
+      preparedStatement.setInt(1, id);
+
+      ResultSet resultSet = preparedStatement.executeQuery();
+
+      while (resultSet.next()) {
+        String first_name = resultSet.getString("first_name");
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: in Java we use ```CamelCase```

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        String last_name = resultSet.getString("last_name");
+        int age = resultSet.getInt("age");
+
+        students.add(new Student(id, first_name, last_name, age));
+      }
+      return students;
+    }
+  }
+
+  public void update(Student student) throws SQLException {
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: here you are updating only age the method name is update

File: src/com/clouway/sqlcommands/StudentRepository.java (hunk 7 above)
Comment (Stanislava88) [inline src/com/clouway/sqlcommands/StudentRepository.java]: done

File: src/com/clouway/sqlcommands/StudentRepository.java (hunk 2 above)
Comment (Stanislava88) [inline src/com/clouway/sqlcommands/StudentRepository.java]: done

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  public Student findByID(int id) throws SQLException {
+    try (PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM students WHERE id=?")) {
+      preparedStatement.setInt(1, id);
+
+      ResultSet resultSet = preparedStatement.executeQuery();
+
+      while (resultSet.next()) {
+        String first_name = resultSet.getString("firstName");
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: still not changed

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        String last_name = resultSet.getString("lastName");
+        int age = resultSet.getInt("age");
+
+        students.add(new Student(id, first_name, last_name, age));
+      }
+      return students;
+    }
+  }
+
+  public void updateAge(Student student) throws SQLException {
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: why are you passing the whole Student obj ,when you need only the age

Comment (Stanislava88) [inline src/com/clouway/sqlcommands/StudentRepository.java]: Student obj is removed

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    repository.register(student2);
+
+    List<Student> actual = repository.findAll();
+    List<Student> expected = Lists.newArrayList(student1, student2);
+
+    assertThat(expected, is(actual));
+  }
+
+  @Test
+  public void findByID() throws Exception {
```
Comment (freekman) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: remove this test it is the same ass happy path

File: src/com/clouway/sqlcommands/StudentRepository.java [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        String lastName = resultSet.getString("lastName");
+        int age = resultSet.getInt("age");
+
+        students.add(new Student(id, firstName, lastName, age));
+      }
+      return students;
+    }
+  }
+
+  public void updateAgeByID(int age, int id) throws SQLException {
```
Comment (freekman) [inline src/com/clouway/sqlcommands/StudentRepository.java]: it would be beter if you swap id with age ```updateAgeByID(int id, int age)```

**Comment by freekman** [comment]
LGTM after  fix comments

File: tests/com/clouway/sqlcommands/StudentRepositoryTest.java (hunk 11 above)
Comment (Stanislava88) [inline tests/com/clouway/sqlcommands/StudentRepositoryTest.java]: done

File: src/com/clouway/sqlcommands/StudentRepository.java (hunk 12 above)
Comment (Stanislava88) [inline src/com/clouway/sqlcommands/StudentRepository.java]: done

**Pull request merged**